psi4/driver/procrouting/dft_funcs/superfuncs.py

```diff
@@ -108,6 +108,33 @@ def build_superfunctional(name, restricted):
     if core.has_option_changed("SCF", "DFT_ALPHA_C"):
         sup[0].set_c_alpha(core.get_option("SCF", "DFT_ALPHA_C"))
 
+    # customization of existing VV10 dispersion:
+    if (core.has_option_changed("SCF", "DFT_DISPERSION_PARAMETERS") and sup[0].vv10_b() > 0.0):
+       nl_tuple = core.get_option("SCF", "DFT_DISPERSION_PARAMETERS")
+       sup[0].set_vv10_b(nl_tuple[0])
+       if len(nl_tuple) > 1:
+          sup[0].set_vv10_c(nl_tuple[1])
+       if len(nl_tuple) > 2:
+          raise ValidationError("too many entries in DFT_DISPERSION_PARAMETERS for DFT-NL")
+    # add VV10 correlation to any functional or modify existing
+    # custom procedures using name 'scf' without any quadrature grid like HF will fail and are not detected
+    if (core.has_option_changed("SCF", "DFT_VV10_B") and core.has_option_changed("SCF", "DFT_VV10_C")):
+        if(name.lower()=='hf'):
+           raise ValidationError("SCF: HF with -NL not implemented")
+        vv10_b = core.get_option("SCF", "DFT_VV10_B")
+        vv10_c = core.get_option("SCF", "DFT_VV10_C")
+        sup[0].set_vv10_b(vv10_b)
+        sup[0].set_vv10_c(vv10_c)
+    elif core.has_option_changed("SCF", "DFT_VV10_B"):
+        if(name.lower()=='hf'):
+           raise ValidationError("SCF: HF with -NL not implemented")
+        vv10_b = core.get_option("SCF", "DFT_VV10_B")
+        sup[0].set_vv10_b(vv10_b)
+        if (abs(sup[0].vv10_c() - 0.0) <= 1e-8):
+            core.print_out("SCF: VV10_C not specified. Using default (C=0.0093)!")
+            sup[0].set_vv10_c(0.0093)
```

Comment:
The empirical dispersion counterpart of this has a rather extensive testing suite in the `dftd3/energy` test. Is this multiple parameter-setting functionality for VV10 also covered in a test?

Reply:
I admit only in a minimal fashion.